src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -531,6 +537,36 @@ private void checkEntityIdMatchesPathIds(TutorialGroup tutorialGroup, Optional<L
         });
     }
 
+    /**
+     * GET /courses/:courseId/tutorial-groups/export : Export tutorial groups for a specific course to a CSV file.
+     *
+     * @param courseId the id of the course for which the tutorial groups should be exported
+     * @param fields   the list of fields to include in the CSV export
+     * @return the ResponseEntity with status 200 (OK) and the CSV file containing the tutorial groups
+     */
+    @GetMapping("courses/{courseId}/tutorial-groups/export")
+    @EnforceAtLeastInstructorInCourse
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<byte[]> exportTutorialGroups(@PathVariable Long courseId, @RequestParam List<String> fields) {
+        var course = courseRepository.findByIdElseThrow(courseId);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        String csvContent = "";
+        try {
+            csvContent = tutorialGroupService.exportTutorialGroupsToCSV(course, user, fields);
+        }
+        catch (IOException e) {
+            throw new BadRequestException("Error occurred while exporting tutorial groups", e);
+        }
+        byte[] bytes = csvContent.getBytes(StandardCharsets.UTF_8);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.TEXT_PLAIN);
```

Comment:
This should be "text/csv" instead of MediaType.TEXT_PLAIN.

Reply:
👍